app/static/js/manageDepartments.js

```diff
@@ -7,8 +7,88 @@ $(document).ready( function(){
     x.DataTable({
         pageLength: 25
     });
+
+    attachModalToDepartment()
+    $('#departmentsTable').on('draw.dt', function() {
+      attachModalToDepartment()
+    })
+    $('#manageDepartmentSupervisorModal').on('hidden.bs.modal', function() {
+      clearDropdown()
+    })
 });
 
+
+function attachModalToDepartment() {
+    $('.supervisorDeptModal').off('click')
+    $('.supervisorDeptModal').on('click', function() {
+      $('#manageSupervisorContent .modal-header .department-name')
+                .replaceWith(`<h2 class='department-name' id="departmentModalSelect" data-department-id="${this.id}">
+                                    ${$(`#${this.id}`).html()}
+                              </h2>`)
+      getSupervisorsInDepartment(this.id)
+    })
+}
+
+
+$("#supervisorModalSelect").on('change', async function() {
+  let supervisor = $('#supervisorModalSelect :selected').val()
+  let department = $('#departmentModalSelect').data('department-id')
+  let data = {"supervisor": supervisor, "department": department}
+  await addSupervisorToDepartment(data)
```

Comment:
This await does not do what you think it does. Recall that the ajax method completes instantaneously after sending out the information to the server. What you want to do is to wait until the data has reached the server successfully but what you're actually doing is only waiting for the ajax to send the information, not to wait until it's been received. This results in an incomplete fetch of the supervisors that should be in the department on the next line since it might exclude the data that we just tried to update. 

To fix this, consider adding a callback parameter to the `addSupervisorToDepartment` function and passing in a function containing everything you'd like to do after the data is received by the server. You can call the callback from the success key-value pair when you make the ajax request.